src/to-be-visible.js

```diff
@@ -14,10 +14,16 @@ function isStyleVisible(element) {
   )
 }
 
+function hasElementAVisibleAttribute(element) {
+  return element.nodeName === 'DETAILS'
+    ? element.hasAttribute('open')
```

Comment:
I'm not sure this will work when the user manually toggles the details panel visibility. This is checking for the hard-coded presence of the attribute, and I think you need to check for the `element.open` property.

I'm basing my assumptions here on the MDN documentation. See [the "Events" section of the `details` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details#Events).

Reply:
I think this is good enough as a MVP. The underlying issue is that we check attributes instead of properties. For both `open` and `hidden` we should consult the DOM properties instead of HTML attributes.

Reply:
> The underlying issue is that we check attributes instead of properties. For both open and hidden we should consult the DOM properties instead of HTML attributes.

That's what I was suggesting. Without it, this selector could give an incorrect result in an actual web page where the details has been toggled open after the initial render.